docs/dotnet/dotnet-aspnet-metrics.md

```diff
@@ -0,0 +1,216 @@
+- [Rate-limiting](#rate-limiting)
+  * [Metric: `aspnet.rate_limiting.active_request_leases`](#metric-aspnetrate_limitingactive_request_leases)
+  * [Metric: `aspnet.rate_limiting.request_lease.duration`](#metric-aspnetrate_limitingrequest_leaseduration)
+  * [Metric: `aspnet.rate_limiting.queued_requests`](#metric-aspnetrate_limitingqueued_requests)
+  * [Metric: `aspnet.rate_limiting.queued_requests.duration`](#metric-aspnetrate_limitingqueued_requestsduration)
+  * [Metric: `aspnet.rate_limiting.rejected_requests`](#metric-aspnetrate_limitingrejected_requests)
+
+<!-- tocstop -->
+
+## Routing
+
+All routing metrics are reported by `Microsoft.AspNetCore.Routing` meter.
+
+### Metric: `aspnet.routing.successful_matches`
+
+**Status**: [Experimental][DocumentStatus]
+
+**TODO: can we try one aspnet.routing.matches** ?
+
+This metric is required.
+
+<!-- semconv metric.aspnet.routing.successful_matches(metric_table) -->
+| Name     | Instrument Type | Unit (UCUM) | Description    |
+| -------- | --------------- | ----------- | -------------- |
+| `aspnet.routing.successful_matches` | Counter | `{match}` | Number of requests that successfully matched to an endpoint. [1] |
+
+**[1]:** Meter name is `Microsoft.AspNetCore.Routing`.
+<!-- endsemconv -->
+
+<!-- semconv metric.aspnet.routing.successful_matches(full) -->
+| Attribute  | Type | Description  | Examples  | Requirement Level |
+|---|---|---|---|---|
+| `aspnet.routing.fallback` | string | TODO | `TODO` | Required |
+| `http.route` | string | The matched route (path template in the format used by the respective server framework). See note below [1] | `/users/:userID?`; `{controller}/{action}/{id?}` | Required |
+
+**[1]:** MUST NOT be populated when this is not supported by the HTTP server framework as the route attribute should have low-cardinality and the URI path can NOT substitute it.
+SHOULD include the [application root](/docs/http/http-spans.md#http-server-definitions) if there is one.
+<!-- endsemconv -->
+
+### Metric: `aspnet.routing.failed_matches`
+
+**TODO: would if be useful to add method name, scheme, or other attributes?**
+
+**Status**: [Experimental][DocumentStatus]
+
+This metric is required.
+<!-- semconv metric.aspnet.routing.failed_matches(metric_table) -->
+| Name     | Instrument Type | Unit (UCUM) | Description    |
+| -------- | --------------- | ----------- | -------------- |
+| `aspnet.routing.failed_matches` | Counter | `{match}` | Number of requests that failed to match to an endpoint. [1] |
+
+**[1]:** An unmatched request may be handled by later middleware, such as the static files or authentication middleware. Meter name is `Microsoft.AspNetCore.Routing`
+<!-- endsemconv -->
+
+<!-- semconv metric.aspnet.routing.failed_matches(full) -->
+<!-- endsemconv -->
+
+## Exception metrics
+
+Metrics reported by `Microsoft.AspNetCore.Diagnostics` meter.
+
+### Metric: `aspnet.diagnostics_handler.exceptions`
```

Comment:
Porting active comment threads from the word doc:

## Liudmila Molkova
seems this metrics is ASP.NET Core specific (and not HTTP-specific) should we call it aspnet.exceptions instead?

July 18, 2023 at 4:46 PM
## Noah Falk
If this one is aspnet, does that mean we also have aspnet.routing.*, aspnet.kestrel.*, aspnet.signalr_http_transport.* ?


I could see a rule that said whenever the technology name is unique (signalr/kestrel) then we don't add aspnet, but I'm not sure users would expect that given we did include AspNetCore in all the .NET namespaces and Meter names.

July 19, 2023 at 4:48 AM
## Liudmila Molkova
> I could see a rule that said whenever the technology name is unique (signalr/kestrel) then we don't add aspnet,That's a great proposal that also aligns well with otel naming recommendations - https://github.com/open-telemetry/opentelemetry-specification/blob/main/specification/common/attribute-naming.md


So kestrel/signalr are root namespaces, routing and rate_limiting are general-purpose terms and should be prefixed with aspnet. How does it sound?

July 19, 2023 at 10:55 AM
## Noah Falk
Do you think that scheme is preferable to one where kestrel and signalr are prefixed by aspnet?


For example in the .NET namespaces and assembly names Kestrel and SignalR don't show up alone, they show up nested under AspNetCore. I wouldn't be surprised if users searched for "aspnet" in their metrics and if that didn't find kestrel or signalr metrics they might not think to do additional searches under those alternate terms.

July 20, 2023 at 1:20 AM
## Noah Falk
Above in the comments Trask was also suggesting it would be appropriate to prefix kestrel and signalr with aspnet. Does that sound good to you @Liudmila Molkova or you believe it is better if they are not prefixed that way?

July 21, 2023 at 5:23 PM
## Liudmila Molkova
As mentioned in the other thread, I don't see any harm or benefit for kestrel (it's a product name already), but signalr should not be tight to asp,net core sinec there hopefully will be client metrics at some point

July 21, 2023 at 6:55 PM

Reply:
In a strict sense ASP.NET isn't limited just to server side components. ASP.NET is also the technology providing client side javascript for SignalR as well. However it does seem fair to say that users are much more likely to associate aspnet with server side technology and would be surprised to see aspnet in client side metrics. There is also an argument that technically SignalR has an open protocol definition so some other entity that isn't asp.net could also implement SignalR and interoperate.

I think these boundaries are fuzzy, but I'd propose kestrel gets prefixed with aspnet and signalr does not.

Reply:
moved kestrel to aspnet.kestrel